sympy/matrices/tests/test_matrices.py

```diff
@@ -3822,3 +3822,27 @@ def test_gramschmidt_conjugate_dot():
     mat = Matrix([[1, I], [1, -I]])
     Q, R = mat.QRdecomposition()
     assert Q * Q.H == Matrix.eye(2)
+
+def test_issue_17827():
+    C = Matrix([
+        [3, 4, -1, 1],
+        [9, 12, -3, 3],
+        [0, 2, 1, 3],
+        [2, 3, 0, -2],
+        [0, 3, 3, -5],
+        [8, 15, 0, 6]
+    ])
+    # Tests for row/col within valid range
+    D = C.elementary_row_op('n<->m', row1=2, row2=5)
+    E = C.elementary_row_op('n->n+km', row1=5, row2=3, k=-4)
+    F = C.elementary_row_op('n->kn', row=5, k=2)
+    assert(D[5, 0] == 0)
+    assert(E[5, 0] == 0)
+    assert(F[5, 0] == 16)
+    # Tests for row/col out of range
```

Comment:
This test should use `raises`. Something like

    raises(ValueError, lambda: C.elementary_row_op('n<->m', row1=2, row2=6))

Reply:
Thanks for reminding.  I modified the tests using `raises`, and I didn't remove the original tests because the bug could lead to misjudging valid row numbers as out of range.